client/src/store/notification/notification.slice.ts

```diff
@@ -0,0 +1,121 @@
+import { apiSlice } from '../api';
+import { ISendNotificationRequestBody, ISubscription } from '@/types';
+
+const notification = apiSlice.injectEndpoints({
+  endpoints: (builder) => ({
+    sendNotification: builder.mutation<void, ISendNotificationRequestBody>({
+      query: (notification: ISendNotificationRequestBody) => ({
+        url: `/push-notification/send-notification`,
+        method: 'POST',
+        body: { ...notification },
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }),
+    }),
+    saveSubscription: builder.mutation<void, ISubscription>({
+      query: ({ id, subscription }) => ({
+        url: '/push-notification/save-subscription',
+        method: 'POST',
+        body: {
+          id,
+          subscription: JSON.stringify(subscription),
+        },
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }),
+    }),
+    unsubscribe: builder.mutation<void, ISubscription>({
+      query: ({ id, subscription }) => ({
+        url: '/push-notification/unsubscribe',
+        method: 'PUT',
+        body: { id, subscription: JSON.stringify(subscription) },
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }),
+    }),
+  }),
+});
+
+export const { useSendNotificationMutation } = notification;
+
+/**
+ * Provides a method for subscribing the user to receive push notifications.
+ * @returns A tuple containing the trigger for initiating the subscription mutation and an object containing response information.
+ *
+ * @see {@link https://redux-toolkit.js.org/rtk-query/usage/mutations}
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/PushManager/subscribe}
+ */
+export const useNotificationSubscriptionMutation = () => {
```

Comment:
For both this function and the one below it, I'm actually not sure if this should be here and described like it is. This is merely wrapping around the actual subscription endpoint to do some validation before it and setting up the service worker before subscribing. The other does something similar. Both these actions feel like a hook to me, but I'm not sure. 

They seem fine, I'm just not sure on the location and naming. You can resolve this with no changes just for documenting.

Reply:
I think I'm going to resolve this for now for simplicity's sake.